From c145fa625f5f7d70fe5d8fda856a30007d242914 Mon Sep 17 00:00:00 2001
From: Joram Schrijver <i@joram.io>
Date: Mon, 5 Aug 2019 15:42:26 +0200
Subject: [PATCH 2/2] Pass through media keys

---
 slock.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/slock.c b/slock.c
index eb07ad4..a0bd7de 100644
--- a/slock.c
+++ b/slock.c
@@ -16,6 +16,7 @@
 #include <sys/types.h>
 #include <X11/extensions/Xrandr.h>
 #include <X11/keysym.h>
+#include <X11/XF86keysym.h>
 #include <X11/Xlib.h>
 #include <X11/Xutil.h>
 
@@ -157,6 +158,16 @@ readpw(Display *dpy, struct xrandr *rr, struct lock **locks, int nscreens,
 			    IsPrivateKeypadKey(ksym))
 				continue;
 			switch (ksym) {
+			case XF86XK_AudioLowerVolume:
+			case XF86XK_AudioMute:
+			case XF86XK_AudioRaiseVolume:
+			case XF86XK_AudioPlay:
+			case XF86XK_AudioPause:
+			case XF86XK_AudioStop:
+			case XF86XK_AudioPrev:
+			case XF86XK_AudioNext:
+				XSendEvent(dpy, DefaultRootWindow(dpy), True, KeyPressMask, &ev);
+				break;
 			case XK_Return:
 				passwd[len] = '\0';
 				errno = 0;
-- 
2.22.0

